Feature/add save and load game

maorharush opened the pull request.

maorharush committed:
Change walls to numWalls on player

maorharush committed:
Add initial game save/load feature

Nom1fan commented on quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/Board.java:
```diff
@@ -141,6 +144,21 @@ public Wall getWall(int x, int y, boolean isHorizontal) {
             return verticalWalls[x][y];
         }
     }
+    public List<Wall> getAllWall(){
```
Suggest to change to `getAllWalls`

maorharush replied:
done

Nom1fan commented on quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/Board.java:
```diff
@@ -141,6 +144,21 @@ public Wall getWall(int x, int y, boolean isHorizontal) {
         }
     }
+    public List<Wall> getAllWall(){
+     List<Wall> wall= new ArrayList<>();
```
suggest to change `wall` to `walls`

Nom1fan commented on quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/Board.java:
```diff
@@ -141,6 +144,21 @@ public Wall getWall(int x, int y, boolean isHorizontal) {
+    public List<Wall> getAllWall(){
+     List<Wall> wall= new ArrayList<>();
+        for (int y=0;y<height;y++){
+            for (int x=0;y<width;x++){
```
Bugfix: change y<width to x<width

maorharush replied:
done

Nom1fan commented on quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/Board.java:
```diff
@@ -141,6 +144,21 @@ public Wall getWall(int x, int y, boolean isHorizontal) {
+     List<Wall> wall= new ArrayList<>();
+        for (int y=0;y<height;y++){
+            for (int x=0;y<width;x++){
+                if(containsWall(x,y,true)){
```
Add method and call once with `true` and once with `false`

maorharush replied:
done, added addWallIfExists() method, by you *-:

Nom1fan commented on quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/GamePersistenceServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public void saveGame(GameSession gameSession){
+        GameRecDBO recorder = new GameRecDBO();
```
Change `recorder` to `gameRecord`

maorharush replied:
done

Nom1fan commented on quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/GamePersistenceServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @Override
+    public void saveGame(GameSession gameSession){
```
Change to `saveTurn` instead of `saveGame`

maorharush replied:
done

maorharush committed:
Cosmetics

maorharush committed:
Add TODOs for changes in table structure + DBO/DAO

maorharush committed:
Add board unit test

maorharush committed:
Add more board tests

maorharush committed:
refactor of GameRecDBO

maorharush committed:
refactor of GameRecDBO

Commit:
Set color of wall to correspond to player

maorharush committed:
small fix to objects names

maorharush committed:
surround setter with if statements

maorharush committed:
added getMaxId method

maorharush committed:
add game_id support

maorharush committed:
add game_id support

maorharush committed:
Merge branch 'develop' into feature/Add-save-and-load-game

# Conflicts:
#	quoridorui/src/main/java/com/harush/zitoon/quoridor/ui/view/MainGame.java

maorharush committed:
Merge remote-tracking branch 'origin/develop' into feature/Add-save-and-load-game

# Conflicts:
#	quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/GameSession.java
#	quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/Player.java
#	quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/Wall.java
#	quoridorui/src/main/java/com/harush/zitoon/quoridor/ui/view/MainGame.java

maorharush committed:
Initial infra for save/load game

maorharush committed:
Merge branch 'develop' into feature/Add-save-and-load-game

maorharush committed:
א

Nom1fan committed:
Fix board test

Nom1fan committed:
Add PlayerAction2GameRecDBOConverter + tests

Nom1fan committed:
Implement save/load feature + test (Still has small bug)

Nom1fan committed:
Fixed small bug

maorharush committed:
implemented getGameRecords() and getPlayerRecords()

Nom1fan committed:
Tidy up test

Nom1fan committed:
Merge remote-tracking branch 'origin/feature/Add-save-and-load-game' into feature/Add-save-and-load-game

Nom1fan committed:
- Update schema of game_records to match DBO
- Implement getAllRecords() API
- Fix bug in GameRecDBO equals method
- Add tests

Nom1fan committed:
- Add game_rec_id primary key to game_records schema
- Fix equals & hashcode in GameRecDBO once and for all
- Add to GameRecDAO getPlayerRecords() API

Nom1fan committed:
Pass new gameID to GameSession

Nom1fan committed:
Fix GameRecDAO getMaxID() API + add test

Nom1fan committed:
Modify DAOFactory to singleton

Nom1fan committed:
Add GameDAO getLastGameID() API + test

Nom1fan committed:
Rename getMaxID() to getLastGameID()

Nom1fan committed:
Add infra for saving player actions

maorharush committed:
add a bunch of docu's to the core

Nom1fan committed:
Improve javadoc

Nom1fan committed:
Major infra changes to support save/load feature (not yet ready)

Nom1fan committed:
Add PopulateBoardUtil test

Nom1fan committed:
Fix GameRecDAO

maorharush committed:
nothing realy

maorharush committed:
nothing realy

Nom1fan committed:
Major infra changes to support save/load feature (not yet ready) #2

Nom1fan committed:
Merge remote-tracking branch 'origin/feature/Add-save-and-load-game' into feature/Add-save-and-load-game

Nom1fan committed:
Add postConstruct() to controller infra

Nom1fan committed:
Update foreign keys in db schema

Nom1fan committed:
Save/Load feature works

Nom1fan committed:
Fix bug in load feature - InitialCoordinate remained

Nom1fan committed:
Fix bug in load feature - Loading correct current player turn
